src/test/java/uk/gov/companieshouse/extensions/api/reasons/ReasonServiceUnitTest.java

```diff
@@ -87,10 +85,16 @@ public void canGetListOfReasons() throws ServiceException {
         reason2.setId("reason2");
         extensionRequestFullEntity.addReason(reason2);
         when(requestsService.getExtensionsRequestById(REQUEST_ID)).thenReturn(Optional.of(extensionRequestFullEntity));
-        when(reasonMapper.entityToDTO(reason1))
-            .thenReturn(mapper.entityToDTO(reason1));
-        when(reasonMapper.entityToDTO(reason2))
-            .thenReturn(mapper.entityToDTO(reason2));
+
+        when(reasonMapper.entityToDTO(any(ExtensionReasonEntity.class))).thenAnswer(invocationOnMock -> {
+            if ((invocationOnMock.getArguments()[0].equals(reason1))) {
+                return mapper.entityToDTO(reason1);
+            } else  if ((invocationOnMock.getArguments()[0].equals(reason2))) {
+                return mapper.entityToDTO(reason2);
+            }
+            return null;
+        });
+
```

Comment:
this approach is the same as using lenient. return null should never be needed.
checkout master branch, run canGetListOfReasons() test, it passes
remove "when(reasonMapper.entityToDTO(reason1)).thenReturn(mapper.entityToDTO(reason1));" - it fails
indicating that reasonMapper.entityToDTO is called with param "reason1"
now in this PR branch the test fails because reasonMapper.entityToDTO is no longer called with reason1 and we haven't changed the code. We need to work out why this is no longer being called rather than just make it optional.

Reply:
i'll try and debug it on mine

Reply:
Let me know what you make of the full error message.

Reply:
I think this thenAnswer block is the correct way to go as it seems that MocitoExtension is more strict than junit runner.
The block however I'd recommend doing as `when(reasonMapper.entityToDTO(any(ExtensionReasonEntity.class))).thenAnswer(invocationOnMock ->
            mapper.entityToDTO(invocationOnMock.getArgument(0))
        );`

Reply:
Found out we just have to reverse the when then -> doreturn when
this works:
doReturn(mapper.entityToDTO(reason1)).when(reasonMapper).entityToDTO(reason1);
doReturn(mapper.entityToDTO(reason2)).when(reasonMapper).entityToDTO(reason2);

Reply:
Updated this to use doReturn so as to avoid the side effect of real method invocation.